File: architecture-examples/gwt/src/com/todo/client/ToDoItem.java
package com.todo.client;

/**
 * An individual ToDo item.
 *
 * @author ceberhardt
 * @author dprotti
 *
 */
public class ToDoItem {

	private String title;

	private boolean completed;

	public ToDoItem(String title) {
		this(title, false);
	}

	public ToDoItem(String title, boolean completed) {
		this.title = title;
		this.completed = completed;
	}

	public boolean isCompleted() {
		return completed;
	}

	public void setCompleted(boolean completed) {
		this.completed = completed;
	}

	public String getTitle() {
		return title;
	}

	public void setTitle(String title) {
		this.title = title;
	}

}

File: examples/angular2_es2015/client/app/components/todo_item.js
'use strict';
import {Component, EventEmitter, Output, Input} from 'angular2/core';
import {TrimPipe} from '../pipes/trim';
import todoItemTemplate from './todo_item.html';

@Component({
	selector: 'todo-item',
	template: todoItemTemplate,
	pipes: [TrimPipe]
})
export class TodoItem {
	@Input()
	todo;

	@Output()
	itemModified = new EventEmitter();

	@Output()
	itemRemoved = new EventEmitter();

	editing = false;

	cancelEditing() {
		this.editing = false;
	}

	stopEditing(editedTitle) {
		this.todo.setTitle(editedTitle.value);
		this.editing = false;

		if (this.todo.title.length === 0) {
			this.remove();
		} else {
			this.update();
		}
	}

	edit() {
		this.editing = true;
	}

	toggleCompletion() {
		this.todo.completed = !this.todo.completed;
		this.update();
	}

	remove() {
		this.itemRemoved.next(this.todo.uid);
	}

	update() {
		this.itemModified.next(this.todo.uid);
	}
}
